### PR TITLE
v5 version of component

### PR DESCRIPTION
**Pull request by moloko**
Also:
* updates to a11y
* amended to use itemsComponentModel as per Accordion, Narrative
* added 'inview' completion as per core presentation components
* get rid of slightly pointless help text in properties.schema

### CODE REVIEW

**Comment by moloko** [comment]
Clearly there's a bit still to do on the tab button styling...!
adapt-tabs

**Comment by moloko** [comment]
fixes #19

**Comment by amygroshek** [comment]
Works. Tested on AT v0.10.3 w/ AF v0.5.4.

**Comment by moloko** [comment]
anyone else feel the need to review/test this - or shall I just merge & release?

**Comment by moloko** [comment]
@kineojen / @guywillis / @kirsty-hames anyone have any time to look at getting the tab buttons styled up a bit better?

**Comment by kirsty-hames** [comment]
Yep I've done this already in a project - I'll pick this up

**Review by kirsty-hames — COMMENTED** [review]
resolves #18

File: js/adapt-tabs.js [hunk 1]
```diff
@@ -1,119 +1,114 @@
...
+        if (tabLayout === 'vertical') {
+          this.setTabLayoutVertical();
+          return;
+        }
+        this.setTabLayoutHorizontal();
+        return;
+      }
+
+      this.$el.addClass('is-horizontal-layout');
+      this.setTabLayoutVertical();
```
Comment (tomgreenfield) [inline js/adapt-tabs.js]: Mismatched layouts on these lines.

Suggest using a ternary for the layout to make it less awkward to read:
```js
var layout = Adapt.device.screenSize === 'large' ?
  this.model.get('_tabLayout') :
  'vertical';

this.$el.addClass('is-' + tabLayout + '-layout');
```

Personally, I would lose the two `setTabLayout` functions and avoid unnecessary setting of 100% width in JS; can instead unset it and fallback to styles added to the `is-vertical-layout` class. 

Could then keep layout stuff in the one place:
```js
setLayout: function () {
  var layout = Adapt.device.screenSize === 'large' ?
    this.model.get('_tabLayout') :
    'vertical';

  var itemWidth = layout === 'horizontal' ?
    100 / this.model.get('_items').length + '%' :
    '';

  this.$el
    .removeClass('is-horizontal-layout is-vertical-layout')
    .addClass('is-' + tabLayout + '-layout');

  this.$('.tabs__nav-item-btn').css('width', itemWidth);
},
```

File: README.md [hunk 2]
```diff
@@ -18,30 +18,24 @@ The attributes listed below are used in *components.json* to configure **Tabs**,
...
-**body** (string): The body text content of the particular item.
+**\_minHeight** (number): This value defines the minimum height the item container displays. By default, the container is responsive to the content height. You only need to set this value if you want to uniform the content container size across tab items.
 
 **instruction** (string): This optional text appears above the component. It is frequently used to
 guide the learner’s interaction with the component.
 
-**_tabLayout** (string): Defines the layout of the tabs. Acceptable values are `left` and `top`. By default, component displays `left` layout for medium screen size and below.
+**\_setCompletionOn** (string): Determines when the component registers as complete. Acceptable values are `"allItems"` and `"inview"`. `"allItems"` (the default) requires the learner to view every single tab; `"inview"` requires only that the component has been viewed (i.e. passed completely through the browser viewport).
+
+**\_tabLayout** (string): Defines the layout of the tabs. Acceptable values are `"left"` and `"top"`. By default, component displays `left` layout for medium screen size and below.
```
Comment (tomgreenfield) [inline README.md]: Needs updating with new layout names.

File: properties.schema [hunk 3]
```diff
@@ -30,15 +30,28 @@
...
-      "default": "left",
-      "title": "Tab button position",
-      "inputType": {"type": "Select", "options":["left", "top"]},
+      "enum": ["horizontal", "vertical"],
+      "default": "horizontal",
+      "title": "Tab button layout",
+      "inputType": {"type": "Select", "options":["vertical", "horizontal"]},
       "validators": ["required"],
-      "help": "Select whether the tab buttons should be shown on the left or or the top of the content panel"
+      "help": "Whether the tab buttons should display horizontal across the top of the content panel, or vertically along the side of the content panel"
```
Comment (tomgreenfield) [inline properties.schema]: Should this mention this only takes effect on large screens?

**Comment by kirsty-hames** [comment]
Thank you for your input @tomgreenfield. I've committed your amends if you have time to review please?

File: properties.schema [hunk 4]
```diff
@@ -30,15 +30,28 @@
...
-      "default": "left",
-      "title": "Tab button position",
-      "inputType": {"type": "Select", "options":["left", "top"]},
+      "enum": ["horizontal", "vertical"],
+      "default": "horizontal",
+      "title": "Tab button layout",
+      "inputType": {"type": "Select", "options":["vertical", "horizontal"]},
       "validators": ["required"],
-      "help": "Select whether the tab buttons should be shown on the left or or the top of the content panel"
+      "help": "Whether the tab buttons should display horizontal across the top of the content panel, or vertically along the side of the content panel. By default, component displays `vertical` layout for medium screen size and below."
```
Comment (tomgreenfield) [inline properties.schema]: Just to point out that backticks <code>`</code> have no effect in the help text.

**Review by tomgreenfield — APPROVED** [review]
:eye:

**Pull request merged**